tests/test_basicio.py

```diff
@@ -269,18 +267,70 @@ def test_du(localserver, cache_expiry, clear_server):
 @pytest.mark.parametrize("cache_expiry", [0, expiry_time])
 def test_glob(localserver, cache_expiry, clear_server):
     remoteurl, localpath = localserver
-    with fsspec.open(remoteurl + "/WalkFolder/testfile1.txt", "wt") as f:
-        f.write(TESTDATA2)
-        f.flush()
-    with fsspec.open(remoteurl + "/WalkFolder/testfile2.txt", "wt") as f:
-        f.write(TESTDATA2)
-        f.flush()
+    os.makedirs(localpath + "/WalkFolder")
+    with open(localpath + "/WalkFolder/testfile1.txt", "w") as fout:
+        fout.write(TESTDATA2)
+    with open(localpath + "/WalkFolder/testfile2.txt", "w") as fout:
+        fout.write(TESTDATA2)
     time.sleep(sleep_time)
     fs, token, path = fsspec.get_fs_token_paths(
         remoteurl, "rt", storage_options={"listings_expiry_time": cache_expiry}
     )
-    print(fs.glob(path[0] + "/*.txt"))
     assert set(fs.glob(path[0] + "/WalkFolder/*.txt")) == {
         path[0] + "/WalkFolder/testfile1.txt",
         path[0] + "/WalkFolder/testfile2.txt",
     }
+
+
+@pytest.mark.parametrize("cache_expiry", [0, expiry_time])
+def test_cat(localserver, cache_expiry, clear_server):
+    remoteurl, localpath = localserver
+    os.makedirs(localpath + "/WalkFolder")
+    with open(localpath + "/WalkFolder/testfile1.txt", "w") as fout:
+        fout.write(TESTDATA1)
+    with open(localpath + "/testfile2.txt", "w") as fout:
+        fout.write(TESTDATA2)
+    time.sleep(sleep_time)
+    fs, token, path = fsspec.get_fs_token_paths(
+        remoteurl, "rt", storage_options={"listings_expiry_time": cache_expiry}
+    )
+    assert fs.cat_file(path[0] + "/testfile2.txt", 4, 9) == b"green"
+
+    paths = [
+        path[0] + "/testfile2.txt",
+        path[0] + "/WalkFolder/testfile1.txt",
+        path[0] + "/testfile2.txt",
+    ]
+    starts = [4, 6, 4]
+    ends = [9, 12, 10]
+    assert fs.cat_ranges(paths, starts, ends, batch_size=100) == [
+        b"green",
+        b"banana",
+        b"green\n",
+    ]
+
+
+def test_make_vectors(localserver, clear_server):
```

Comment:
this can just be `def test_make_vectors():`

Reply:
ok